utils.go

```diff
@@ -0,0 +1,18 @@
+package nmap
+
+// https://siongui.github.io/2018/04/14/go-remove-duplicates-from-slice-or-array/
+func RemoveDuplicatesFromSlice(s []string) []string {
+	m := make(map[string]bool)
+	for _, item := range s {
+		if _, ok := m[item]; ok {
```

Comment:
Instead of

```go
		if _, ok := m[item]; ok {
		} else {
			m[item] = true
		}
```

Please use the form

```go
		if _, ok := m[item]; !ok {
			m[item] = struct{}{}
		}
```